src/java/htsjdk/samtools/metrics/MetricBase.java

```diff
@@ -84,7 +85,10 @@ public int hashCode() {
         int result = 0;
         for (final Field f : getClass().getFields()) {
             try {
-                result = 31 * result + f.get(this).hashCode();
+                Object value = f.get(this);
+                value = new FormatUtil().format(value); //format the value the way it will be written to disk
```

Comment:
That's horrifying... why does `DateFormat` have state?

We should add a comment to `FormatUtil` explaining that then.  It looks like it's just needlessly allocating objects.